### PR TITLE
Move fetching of dataset value down into VisDisplay

### PR DESCRIPTION
**Pull request by axelboc**

### CODE REVIEW

File: src/h5web/dataset-visualizer/DatasetVisualizer.tsx [hunk 1]
```diff
@@ -66,30 +29,20 @@ function DatasetVisualizer(props: Props): ReactElement {
...
-              mapperState: VIS_DEFS[vis].defaultMappingState(dataset),
-            });
-          }}
+          onChange={setActiveVis}
         />
         {VisToolbar && <VisToolbar />}
       </div>
       <div className={styles.displayArea}>
         <VisDisplay
+          key={`${dataset.id}_${activeVis}`}
```
Comment (axelboc) [inline src/h5web/dataset-visualizer/DatasetVisualizer.tsx:38]: This forces a reset of the state of `VisDisplay` (dataset value + mapper state) whenever the visualization and/or dataset changes.

File: src/h5web/dataset-visualizer/DatasetVisualizer.tsx [hunk 2]
```diff
@@ -1,63 +1,26 @@
...
 interface Props {
   dataset: HDF5Dataset;
 }
 
-interface State {
-  activeVis: Vis;
-  value?: HDF5Value;
-  loading?: boolean;
-  mapperState?: DimensionMapping;
-  prevDataset?: HDF5Dataset;
```
Comment (axelboc) [inline src/h5web/dataset-visualizer/DatasetVisualizer.tsx:21]: The derived state condition below is much simpler, and we no longer need to store the `prevDataset`.

File: src/h5web/dataset-visualizer/DatasetVisualizer.tsx (hunk 1 above)
Comment (loichuder) [reply src/h5web/dataset-visualizer/DatasetVisualizer.tsx:38]: I would rather see this in comment in the code :wink;

**Review by loichuder — APPROVED** [review]
:rocket:

File: src/h5web/dataset-visualizer/DatasetVisualizer.tsx (hunk 1 above)
Comment (axelboc) [reply src/h5web/dataset-visualizer/DatasetVisualizer.tsx:38]: Oh sorry, I did mean to 😓

**Pull request merged**